skbeam/core/accumulators/binned_statistic.py

```diff
@@ -0,0 +1,505 @@
+            Empty bins will be represented by NaN.
+          * 'median' : compute the median of values for points within each
+            bin. Empty bins will be represented by NaN.
+          * 'count' : compute the count of points within each bin.  This is
+            identical to an unweighted histogram.  `values` array is not
+            referenced.
+          * 'sum' : compute the sum of values for points within each bin.
+            This is identical to a weighted histogram.
+          * function : a user-defined function which takes a 1D array of
+            values, and outputs a single numerical statistic. This function
+            will be called on the values in each bin.  Empty bins will be
+            represented by function([]), or NaN if this returns an error.
+
+    bins : int or [int, int] or array_like or [array, array], optional
+        The bin specification:
+
+          * the number of bins for the two dimensions (nx=ny=bins),
+          * the number of bins in each dimension (nx, ny = bins),
+          * the bin edges for the two dimensions (x_edges = y_edges = bins),
+          * the bin edges in each dimension (x_edges, y_edges = bins).
+
+    range : (2,2) array_like, optional
+        The leftmost and rightmost edges of the bins along each dimension
+        (if not specified explicitly in the `bins` parameters):
+        [[xmin, xmax], [ymin, ymax]]. All values outside of this range will be
+        considered outliers and not tallied in the histogram.
+
+    See Also
+    --------
+    numpy.histogram2d, binned_statistic, binned_statistic_dd
+
+    """
+
+    def __init__(self, x, y, statistic='mean',
+                 bins=10, range=None):
+        # This code is based on np.histogram2d
+        try:
+            N = len(bins)
+        except TypeError:
+            N = 1
+
+        if N != 1 and N != 2:
+            xedges = yedges = np.asarray(bins, float)
+            bins = [xedges, yedges]
+
+        super(BinnedStatistic2D, self).__init__([x, y], statistic=statistic,
+                                                bins=bins, range=range)
+
+    def __call__(self, values):
+        return super(BinnedStatistic2D, self).__call__(values)
+
+
+class RPhiBinnedStatistic(BinnedStatistic2D):
+    """
+    Create a 2-dimensional histogram by binning a 2-dimensional
+    image in both radius and phi.
+    """
+
+    def __init__(self, rbins, phibins, xsize, ysize, cartesian,
+                 xc=None, yc=None, rrange=None, phirange=None, mask=None,
+                 statistic='mean'):
+        """
+        Parameters:
+        -----------
+        rbins: int
+            number of radial bins in returned histogram.
+        phibins: int
+            number of phi bins in returned histogram.
+        xsize,ysize: int
+            shape of image in pixels.  see "cartesian" parameter
+            for definition of x/y.
+        cartesian: bool, optional
```

Comment:
This seems like it will cause more trouble than it is worth.  I would like to adopt the skimage convention and use (row, col) for docs and variable names.